source/common/upstream/health_discovery_service.cc

```diff
@@ -128,17 +128,10 @@ envoy::service::health::v3::HealthCheckRequestOrEndpointHealthResponse HdsDelega
           if (host->health() == Host::Health::Healthy) {
             endpoint->set_health_status(envoy::config::core::v3::HEALTHY);
           } else {
-            switch (host->getActiveHealthFailureType()) {
-            case Host::ActiveHealthFailureType::TIMEOUT:
+            if (host->healthFlagGet(Host::HealthFlag::ACTIVE_HC_TIMEOUT)) {
               endpoint->set_health_status(envoy::config::core::v3::TIMEOUT);
-              break;
-            case Host::ActiveHealthFailureType::UNHEALTHY:
-            case Host::ActiveHealthFailureType::UNKNOWN:
+            } else {
               endpoint->set_health_status(envoy::config::core::v3::UNHEALTHY);
```

Comment:
Is this right? What if none of the flags are set?

Reply:
I believe this is the same behavior as before? Mark a special flag for timeout otherwise mark unhealthy?

Reply:
Oh I see I missed the outer if block that means this only runs when the host is unhealthy. Seems right to me then